File: solutions/uri/1281/1281.cpp
#include <cstdio>
#include <iostream>
#include <map>
#include <string>

int main() {
    double price, result;
    int n, p, q, quantity;
    std::map< std::string, double > prices;
    std::string fruit;

    std::cin >> n;

    while (n--) {
        result = 0.0;
        prices.clear();

        std::cin >> p;

        while (p--) {
            std::cin >> fruit >> price;
            prices[fruit] = price;
        }

        std::cin >> q;

        while (q--) {
            std::cin >> fruit >> quantity;
            result += prices[fruit] * quantity;
        }

        printf("R$ %.2lf\n", result);
    }

    return 0;
}

File: solutions/beecrowd/1281/1281.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

n = int(input())

for _ in range(n):
    result = 0.0
    prices = {}

    p = int(input())

    for _ in range(p):
        fruit, price = input().split()
        price = float(price)
        prices[fruit] = price

    q = int(input())

    for _ in range(q):
        fruit, quantity = input().split()
        quantity = int(quantity)
        result += prices[fruit] * quantity

    output(f'R$ {result:.2f}\n')
